### ISSUE TITLE
Trying to copy a group results in incorrect pasting

### ISSUE THREAD

**Issue by AtsusaKaneytza** [mypaint/mypaint#320]
I'm using Windows 8.1, MyPaint 1.2.0-alpha.20150504+git.a80a54b, downloaded from Tumagonx's blog.

Something i noticed while painting, and I tried to do a "copy/paste" operation
(had to resort to duplicating/normalizing the layer I was trying to paste instead, so no impact on my art at least)
I'm doing my best to try and reproduce this issue with "new" drawings, but I'm having a hard time trying. This is the artwork I was working on at the time:
![image](https://cloud.githubusercontent.com/assets/8382492/7737652/4a6525e2-ff12-11e4-9e9b-3a1dca774123.png)

This is how the layer I'm trying to copy looks by itself:
![image](https://cloud.githubusercontent.com/assets/8382492/7737683/a5ad689c-ff12-11e4-93bf-c642bd1075a3.png)

This is how it pastes when you isolate the layer:
![image](https://cloud.githubusercontent.com/assets/8382492/7737687/bb350620-ff12-11e4-9552-4c27641201b4.png)

Which is... well, as you can see, a very erroneous paste. Lots of pixels are missing, and there are some stray, noisy pixels that shouldn't be there.
![image](https://cloud.githubusercontent.com/assets/8382492/7737715/f2f31ebc-ff12-11e4-9194-ff5c6ca43137.png)

I've tried reproducing this with a new scrap drawing, to see if I could achieve the same effect:
![image](https://cloud.githubusercontent.com/assets/8382492/7737746/3f8ea372-ff13-11e4-80c5-22020e8dbc08.png)

No dice.
Tried it with a Group-Within-A-Group
![image](https://cloud.githubusercontent.com/assets/8382492/7737844/1ce69f04-ff14-11e4-9544-88f28295b766.png)

Also does not work.
If you look at the Layers subwindow in the first artwork, you can see it's very layer heavy. Lots of the layers in the first one are have various Layer modes, and I tried using a bunch of layer modes in the second one but that doesn't work either.

So yeah... I'm not sure if anyone can reproduce this. :/

(I did also discover another bug in the midst of this, concering pasting a layer and it ends up elsewhere, but that's a different bug report)

**Comment by martinxyz** [comment]
Try throwing in some undo-redo (or undo only), for both painting actions and changes to the layer stack; this may help to reproduce this.

**Comment by AtsusaKaneytza** [comment]
@martinxyz I'm not sure what you mean, could you elaborate further please?

**Comment by AtsusaKaneytza** [comment]
Okay, this happened again, but with a different artwork. I think I managed to pinpoint the problem though
![bug 1](https://cloud.githubusercontent.com/assets/8382492/7786084/06405700-017a-11e5-9d8d-aac7b63a6334.png)
![bug 2](https://cloud.githubusercontent.com/assets/8382492/7786085/06509782-017a-11e5-9da4-ea0027dee28e.png)
![bug 3](https://cloud.githubusercontent.com/assets/8382492/7786087/065237cc-017a-11e5-9094-beebcfca1af5.png)
![bug 4](https://cloud.githubusercontent.com/assets/8382492/7786086/06518386-017a-11e5-8bc1-6860ebd19d17.png)

So basically, if there's a layer with no pixels in it in a group and you try to copy/paste that group, it will wind up messed up like I experienced previously.

**Comment by jplloyd** [comment]
Not sure when this was fixed, but I cannot reproduce this in the latest release.
If this is still a problem, make a comment and we'll reopen the issue.
